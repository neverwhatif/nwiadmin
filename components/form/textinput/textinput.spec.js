import React from 'react';
import { shallow } from 'enzyme';

import TextInput from './index';

describe('TextInput', () => {
    it('should render', () => {
        expect(shallow(<TextInput onChange={() => null} />)).toMatchSnapshot();
    });
    it('should handle change', () => {
        let number = 0;
        const component = shallow(<TextInput onChange={() => { number += 1; }} />);

        expect(number).toBe(0);
        component.simulate('change');
        expect(number).toBe(1);
    });
});
