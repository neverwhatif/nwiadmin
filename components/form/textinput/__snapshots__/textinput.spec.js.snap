// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    hasError={false}
    name=""
    onChange={[Function]}
    placeholder={null}
    readOnly={false}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "root",
      "id": "",
      "name": "",
      "onChange": [Function],
      "placeholder": null,
      "readOnly": false,
      "value": "",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "root",
        "id": "",
        "name": "",
        "onChange": [Function],
        "placeholder": null,
        "readOnly": false,
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
